### ISSUE TITLE
Programatically Setting Start and End Date Does Not work

### ISSUE THREAD

**Issue by isaac2004** [nkalinov/ng2-datetime#105]
`  <datetime [(ngModel)]="selectedPeriod.startDate"
                                              required (ngModelChange)="validateStart()"
                                              [timepicker]="false" [datepicker]="datepickerOptions"
                                              class="form-control" placeholder="Start Date"></datetime>

   <datetime [(ngModel)]="selectedPeriod.endDate" formControlName="selectedEndDate"
                                              required (ngModelChange)="validateEnd()"
                                              [timepicker]="false" [datepicker]="datepickerOptions"
                                              class="form-control" placeholder="End Date"></datetime>`

`  validateStart(): void {
        this.datepickerOptions.startDate = this.selectedPeriod.startDate;
    }

  validateEnd(): void {
        this.datepickerOptions.endDate = this.selectedPeriod.endDate;
    }`
Doing above does not work as expected, I will be able to select dates that do not fit the new datepicker options

**Comment by vsanat** [comment]
The value which you are trying to set must be of type "Date".
If by chance you are trying to set string value then it won't accept that.

**Comment by nkalinov** [comment]
You have several problems with your code.

1. You have a custom `ngModelChange` handler but also `[(ngModel)]` which is already doing it. I'm not sure what the behavior is, I guess you are overriding it, but please leave only `[ngModel]`.

2. When you specify a custom `ngModelChange` handler you have to take care of the model value update.
So change: 
- your template: `(ngModelChange)="validateStart($event)"` 
- and the handler (**note the comments!**): 
````
validateStart(value: Date): void {
    this.selectedPeriod.startDate = value; // update the model

    // do not mutate the object or angular won't detect the changes
    // object spread is available since TS 2.1, else use Object.assign
    this.datepickerOptions = {
        ...this.datepickerOptions,
        startDate: this.selectedPeriod.startDate;
    }
}
````
- do the same for `validateEnd()`
